$OpenBSD: patch-libavcodec_aacenc_tns_h,v 1.1 2015/09/15 12:49:42 ajacoutot Exp $

aacenc_tns: redo coefficient quantization and decision making

aacenc_tns: readjust values for new TNS decision making

--- libavcodec/aacenc_tns.h.orig	Sat Sep 12 15:01:35 2015
+++ libavcodec/aacenc_tns.h	Sat Sep 12 15:02:13 2015
@@ -33,9 +33,12 @@
 /* Could be set to 3 to save an additional bit at the cost of little quality */
 #define TNS_Q_BITS 4
 
+/* Coefficient resolution in short windows */
+#define TNS_Q_BITS_SHORT 3
+
 /* TNS will only be used if the LPC gain is within these margins */
-#define TNS_GAIN_THRESHOLD_LOW  1.395f
-#define TNS_GAIN_THRESHOLD_HIGH 11.19f
+#define TNS_GAIN_THRESHOLD_LOW  1.437f
+#define TNS_GAIN_THRESHOLD_HIGH 21.19f
 
 /* If the energy ratio between the low SFBs vs the high SFBs is not between
  * those two values, use 2 filters instead */
@@ -43,7 +46,7 @@
 #define TNS_E_RATIO_HIGH 1.23
 
 /* Do not use TNS if the psy band spread is below this value */
-#define TNS_SPREAD_THRESHOLD 37.081512f
+#define TNS_SPREAD_THRESHOLD 0.5f
 
 void ff_aac_encode_tns_info(AACEncContext *s, SingleChannelElement *sce);
 void ff_aac_apply_tns(AACEncContext *s, SingleChannelElement *sce);
